aws/sdk/integration-tests/s3/tests/endpoints.rs

```diff
@@ -33,27 +40,103 @@ async fn virtual_hosted_buckets() {
 
 #[tokio::test]
 async fn force_path_style() {
-    let (conn, captured_request) = capture_request(None);
-    let sdk_config = SdkConfig::builder()
-        .credentials_provider(SharedCredentialsProvider::new(Credentials::new(
-            "ANOTREAL",
-            "notrealrnrELgWzOk3IfjzDKtFBhDby",
-            Some("notarealsessiontoken".to_string()),
-            None,
-            "test",
-        )))
-        .region(Region::new("us-west-4"))
-        .http_connector(conn.clone())
-        .build();
-    let force_path_style =
-        Client::from_conf(Builder::from(&sdk_config).force_path_style(true).build());
-    let _ = force_path_style
-        .list_objects_v2()
-        .bucket("test-bucket")
-        .send()
-        .await;
+    let (captured_request, client) = test_client(|b| b.force_path_style(true));
+    let _ = client.list_objects_v2().bucket("test-bucket").send().await;
     assert_eq!(
         captured_request.expect_request().uri().to_string(),
         "https://s3.us-west-4.amazonaws.com/test-bucket/?list-type=2"
     );
 }
+
+#[tokio::test]
+async fn fips() {
+    let (captured_request, client) = test_client(|b| b.use_fips(true));
+    let _ = client.list_objects_v2().bucket("test-bucket").send().await;
+    assert_eq!(
+        captured_request.expect_request().uri().to_string(),
+        "https://test-bucket.s3-fips.us-west-4.amazonaws.com/?list-type=2"
+    );
+}
+
+#[tokio::test]
+async fn dual_stack() {
+    let (captured_request, client) = test_client(|b| b.use_dual_stack(true));
+    let _ = client.list_objects_v2().bucket("test-bucket").send().await;
+    assert_eq!(
+        captured_request.expect_request().uri().to_string(),
+        "https://test-bucket.s3.dualstack.us-west-4.amazonaws.com/?list-type=2"
+    );
+}
+
+#[tokio::test]
+async fn multi_region_access_points() {
+    let (_captured_request, client) = test_client(|b| b);
+    let response = client
+        .get_object()
+        .bucket("arn:aws:s3::123456789012:accesspoint/mfzwi23gnjvgw.mrap")
+        .key("blah")
+        .send()
+        .await;
+    let error = response.expect_err("should fail—sigv4a is not supported");
+    assert!(
+        dbg!(format!("{:?}", error)).contains("No auth schemes were supported"),
+        "message should contain the correct error, found: {:?}",
+        error
+    );
+}
+
+#[tokio::test]
+async fn s3_object_lambda() {
+    let (captured_request, client) = test_client(|b| b);
+    let _ = client
+        .get_object()
+        .bucket("arn:aws:s3-object-lambda:us-east-100:123412341234:accesspoint/myolap")
+        .key("s3.txt")
+        .customize()
+        .await
+        .unwrap()
+        .map_operation(|mut op| {
+            op.properties_mut()
+                .insert(UNIX_EPOCH + Duration::from_secs(1234567890));
+            Result::<_, Infallible>::Ok(op)
```

Comment:
Is there anything we can do to the signature of `map_operation` to avoid the need for being explicit when returning `Ok`

Reply:
we can make another function that doesn't need to accept a result